app/models/user_course_subject.rb

```diff
@@ -7,9 +7,36 @@ class UserCourseSubject < ApplicationRecord
 
   after_create :add_user_tasks
 
+  scope :order_create, ->{order(created_at: :asc)}
+  scope :prev_subject, (lambda do |cur|
+    where(user_course_id: cur.user_course_id)
+    .where("id < ?", cur.id).order(id: :desc).first
+  end)
+
   enum status: {start: 0, inprogress: 1, finished: 2, canceled: 3}
 
-  private
+  def pre_sub_finished?
+    return true if id == user_course.user_course_subjects.first.id
+
+    prev_sub = UserCourseSubject.prev_subject self
+    prev_sub.finished?
+  end
+
+  def current_start_time
+    return start_time unless start_time.nil?
```

Comment:
```suggestion
    return start_time if start_time
```
Sao dùng 2 lần phủ định thế em